### PR TITLE
Two-Stage Admission KEP: Rename QuotaBookingMode to PreemptionPolicy

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Renames QuotaBookingMode to PreemptionPolicy, as the previous name was conflicting with the condition `QuotaReserved`.

The new API is focused on the user visible effect: preemption. It also gives more clarity about how to interpret QuotaReserved and what to do when multiple AdmissionChecks have different PreemptionPolicies.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/assign @mwielgus 
cc @trasc

File: keps/993-two-phase-admission/README.md [hunk 1]
```diff
@@ -163,22 +163,20 @@ type AdmissionCheckSpec struct {
...
-// Definitons of QuotaBookingModes.
-// BookAfterQuotaCheck is more restrictive than NoBooking.
 const (
-	// No booking, unless a check explicitely asks for it reporting quota exhausted.
-	NoBooking QuotaBookingMode = "NoBooking"
-	
-	// Book quota (and preempt) immediately after the basic quota checks 
-	// are done, without waiting for the additional checks to cpmplete.
-	BookAfterQuotaCheck QuotaBookingMode = "BookAfterQuotaCheck"
+	NoBooking PreemptionPolicy = "Anytime"
```
Comment (mwielgus) [inline keps/993-two-phase-admission/README.md:178]: Match name with the value. Here and below.

Comment (alculquicondor) [reply keps/993-two-phase-admission/README.md]: Done

**Comment by trasc** [comment]
LGTM

**Pull request merged**